lib/recipes-data/src/lib/transform.ts

```diff
@@ -1,84 +1,109 @@
-import { FeaturedImageWidth, ImageDpr, PreviewImageWidth } from './config';
-import type { Contributor, RecipeImage } from './models';
-import { extractCropDataFromGuimUrl } from './utils';
+import type {Sponsorship} from "@guardian/content-api-models/v1/sponsorship";
+import {SponsorshipType} from "@guardian/content-api-models/v1/sponsorshipType";
+import {FeaturedImageWidth, ImageDpr, PreviewImageWidth} from './config';
+import type {Contributor, RecipeImage} from './models';
+import {extractCropDataFromGuimUrl} from './utils';
+
+export type RecipeTransformationFunction = (recipeData: Record<string, unknown>) => Record<string, unknown>;
 
 const getFastlyUrl = (
-	imageId: string,
-	cropId: string,
-	dpr: number,
-	desiredWidth: number,
-	originalWidth: number,
+  imageId: string,
+  cropId: string,
+  dpr: number,
+  desiredWidth: number,
+  originalWidth: number,
   extension: string
 ) => `https://i.guim.co.uk/img/media/${imageId}/${cropId}/master/${originalWidth}.${extension}?width=${desiredWidth}&dpr=${dpr}&s=none`;
 
 export const replaceFastlyUrl = (
-	recipeId: string,
-	image: RecipeImage,
-	desiredWidth: number,
-	dpr: number,
+  recipeId: string,
+  image: RecipeImage,
+  desiredWidth: number,
+  dpr: number,
 ): RecipeImage => {
   const cropData = extractCropDataFromGuimUrl(image.url);
 
-	if (!cropData) {
-		console.warn(
-			`Error adding fastly URL to recipe ${recipeId} - no crop data found for image with url ${image.url}.`,
-		);
-		return image;
-	}
+  if (!cropData) {
+    console.warn(
+      `Error adding fastly URL to recipe ${recipeId} - no crop data found for image with url ${image.url}.`,
+    );
+    return image;
+  }
 
-	const { mediaId, cropId, width, extension } = cropData;
+  const {mediaId, cropId, width, extension} = cropData;
 
-	return {
-		...image,
-		url: getFastlyUrl(mediaId, cropId, dpr, desiredWidth, width, extension),
-	};
+  return {
+    ...image,
+    url: getFastlyUrl(mediaId, cropId, dpr, desiredWidth, width, extension),
+  };
 };
 
 export type RecipeWithImageData = {
-	id: string;
-	featuredImage: RecipeImage | string; // the latter is an old image format that appears in our test fixtures
-	previewImage?: RecipeImage | string;
+  id: string;
+  featuredImage: RecipeImage | string; // the latter is an old image format that appears in our test fixtures
+  previewImage?: RecipeImage | string;
+};
+
+const getSponsorshipTypeNames: Record<number, string> = {
+  [SponsorshipType.SPONSORED]: "Sponsored",
+  [SponsorshipType.FOUNDATION]: "Foundation",
+  [SponsorshipType.PAID_CONTENT]: "Paid content"
 };
```

Comment:
It's a shame we can't get these strings out of the Thrift models :disappointed:

Reply:
I had a look through the typescript handbook, and I think we can, actually! Every enum apparently gets a [reverse mapping](https://www.typescriptlang.org/docs/handbook/enums.html#reverse-mappings) from its values to the value names, which we could then postprocess ourselves to get the casing we want. The following seems to work as desired [in the typescript playround](https://www.typescriptlang.org/play/?ssl=35&ssc=1&pln=39&pc=1#code/KYOwrgtgBAygDgexAZwQJ2QCwJZwCoCecwUA3gFBRWwAKA8gHIx0BKAogCJQC8UADABpK1AGJ0Aqgw4BBPAElGPKAEYh1KDWlyOAfQDCjPGwZ4lAJnIBfcuQAuREgGlgBZEoDWLhADMo94j6wiCjoWLiExDYAxkioADbAAHRxCADmABT+wIHwsaE4+A4AlOQxIQnJaem5IRgFEcAA2jWodeEOiTD0TKycALol0bG2UKnAti357cQMAIYQwMgAXFAswDFoACYAPJNthcQCUMi2aNggqQB8ShRUzcGtYQdJXYzM7Bx9KwBEe8Cb3zUUHueX2DUSYkkMnkjC+UG+IgQYBAm1mtmwSEBwhBtSe4M02n0hmMeDh3xos2wmygZVsoFs3ysAG4bN5kVF0UhRuM5gt0icVns8cUViczhcyMI0OMwGgQEFQcLiI0Tn0rENykkUhkAAZjWy84DVbrvThFJQAElI+sN1QeU2enRNvQ4RUsOpKZXiWqqep58yNkKksgUDHNvCtNoDdsV9Q6QehobdHtKeQq2vSfoN0YJugMJhJ4agkf9fKFceIiVzRILJmTgzZIA5GPlNuwUWAtoFCtxFeARVFp3OqUlVGltll8vL0yaqsStgQABkEAB3YBoPSzZBGoos6ypzWVXVtju2149D5FksG9ud6PTx3n02u92etM+4882+2hMhxhX60v1Pe97TBeMJGDGEw1fA9vSPTMTzvPlq3zIw60tQCb2AstQKVJIUOJOsYPIRtmy5fUADV1wIBhv2jbsHwaAdjiHCVbhpYYoBSNc0CURiHBVWw+nnJdV3XTdt3SXcpRlOUoB1K1uPXRo+GEhdxDgYgNy3HdLEUsS0ESZA4lvdJlCKRJpTgOJZg7dJvh0QF4XhZN1VghB019SjqNo7CjSfF0AO8tAaLonDYxnJ03kCmCvQ8j8EPGKiQt8pDAwgxN-ww4LQr8mNe0i38oPrdzPM-WxktytL0gI2s8CCpKfLC-zcL7KstDzQj6pgoA):

```typescript
function getNiceName(st: SponsorshipType): string {
  const lower = SponsorshipType[st].toLowerCase();
  return `${lower[0].toUpperCase()}${lower.slice(1).replace("_", " ")}`
}
```

Reply:
😁 @Divs-B would you mind having a go at @emdash-ie 's suggestion? I can pair with you if it gets tricky but I think it should just be a case of inserting the above function and replacing the lookup table with it?

the change should get picked up in the tests I hope.

Reply:
(though we should deal with `SponsorshipType[st]` returning `undefined`, just in case the upstream models add a new value and we don't get updated)

Reply:
Thanks @emdash-ie , I did try checking this by doing  `SponsorshipType[SponsorshipType[st]]` which is giving string literal of that, so yes reverse mapping definetly can be useful. I applied your code for string manipulation as well in exisitng code and included @fredex42 suggestion (below) for having "Unknown" sposnorship type if not found.

Reply:
> 😁 @Divs-B would you mind having a go at @emdash-ie 's suggestion? I can pair with you if it gets tricky but I think it should just be a case of inserting the above function and replacing the lookup table with it?
> 
> the change should get picked up in the tests I hope.


Thanks @fredex42 for asking, its done now and should work.